### DISCUSSION TITLE
File Association App LifeCycle File Handling

### DISCUSSION THREAD

**Discussion by WebAssemblyMan** [microsoft/WindowsAppSDK#2089] [Q&A] [answered]
Hi,

I am working with File Association and File handling for a WindowsSDK App.

In App.xaml.cs OnLaunched, I have the following, may I know how do I get the file that triggered the app?

        protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
        {
            var activatedEventArgs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
            if (activatedEventArgs.Kind == Microsoft.Windows.AppLifecycle.ExtendedActivationKind.File)
            {
                System.Diagnostics.Debug.WriteLine("Launch new instance to hadle file?");
                //how to get the file or file name here? 
                //In UWP I can use System.Diagnostics.Debug.WriteLine(args.Files[0].Name);
            }
        }

Package.appxmanifest

      <Extensions>
        <uap:Extension Category="windows.fileTypeAssociation">
          <uap:FileTypeAssociation Name="lxml">
            <uap:SupportedFileTypes>
              <uap:FileType>.lxml</uap:FileType>
            </uap:SupportedFileTypes>
          </uap:FileTypeAssociation>
        </uap:Extension>
      </Extensions>

**Comment by seven-mile** [comment, marked as answer]
I think you can cast `activatedEventArgs` into `Windows::ApplicationModel::Activation::IFileActivatedEventArgs` interface directly. Then access properties in it to get the information you need.

**Reply by WebAssemblyMan** [reply]
Hi, 
@seven-mile 

Thanks for your reply.

In my C# app, I am getting

CS0030	Cannot convert type 'Microsoft.Windows.AppLifecycle.AppActivationArguments' to 'Windows.ApplicationModel.Activation.IFileActivatedEventArgs'

**Reply by andreww-msft** [reply]
AppActivationArguments exposes a Data property - that's the thing you can cast:

AppActivationArguments args = AppInstance.GetCurrent().GetActivatedEventArgs();
ExtendedActivationKind kind = args.Kind;
if (kind == ExtendedActivationKind.File)
{
    try
    {
        if (args.Data is IFileActivatedEventArgs fileArgs)
        {
            IStorageItem file = fileArgs.Files[0];
        }
    }
}

See the [C# WinUI instancing sample](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Samples/AppLifecycle/Instancing/cs-winui-packaged/CsWinUiDesktopInstancing/CsWinUiDesktopInstancing/Program.cs) for more details.

**Reply by WebAssemblyMan** [reply]
Hi @seven-mile, @andreww-msft 

I get it now. Thank you for your help.
